packages/autocomplete-core/src/types/AutocompleteOptions.ts

```diff
@@ -176,6 +177,15 @@ export interface AutocompleteOptions<TItem extends BaseItem> {
    * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-reshape
    */
   reshape?: Reshape<TItem>;
+  /**
+   * Whether to enable the Insights plugin and load the Insights library if it has not been loaded yet.
+   *
+   * See [**autocomplete-plugin-algolia-insights**](https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-plugin-algolia-insights/) for more information.
+   *
+   * @default false
+   * @link https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-insights
+   */
+  insights?: CreateAlgoliaInsightsPluginParams | boolean;
```

Comment:
We'll probably need to explain in the documentation that setting this to `false` doesn't impact the behavior of an Insights plugin explicitly added by the customer.

Reply:
Good point, not sure if we already have a task to add this new option to the docs or if it's part of [FX-2249](https://algolia.atlassian.net/browse/FX-2249)

Reply:
It probably can be done in the task related to the doc.

Reply:
shouldn't it remove the default one if it's set by the user, like in InstantSearch? I think it's less surprising

Reply:
@Haroenv You mean, setting `insights=false` would disable any Insights middleware, internal or manually added?

Reply:
Oh, I forgot we have this behavior in InstantSearch. My thoughts are that if a customer explicitly makes the effort of adding an Insights plugin, it should be the one used instead of the one that is created behind a black box.

Reply:
No, I mean that setting an insights plugin yourself overrides the insights plugin (options) set by "insights: boolean"

Reply:
@Haroenv That's the current behavior: https://github.com/algolia/autocomplete/pull/1121/files#diff-c854a8d822f38661738b9ae32cc94ad7a37086c349a638886aa520225a80b81aR150-R169